### PR TITLE
zarajoy/c01-git03

### PR DESCRIPTION
**Pull request by zarajoy**
# Description

Exercise: c01-git03 

Student Name: zarajoy

> exercise c01-git03 exercises and answers.

## Type of change

- [X] Exercise Submission (one exercise per PR)
  - [X] This PR is just for **one** exercise of a class
  - [X] PR name follows the pattern `<github-username>/<exercise-number>`
  - [X] Added exercise files inside folder `/classes/<class- number>/exercises/<exercise-number>/<github-username>/`
	- [ ] Review [exercise submission steps](./README.md#Exercises)
- [ ] Class content (instructors/admins)
- [ ] Documentation update
- [ ] Repo management

### CODE REVIEW

File: classes/01class/exercises/c01-git03/zarajoy/c01-git03.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+1. Did you notice any difference in the size of the repo before and after adding the big file?
+Yes. The repo size was more than what I expected after I added the big file (eg increase in size was more than the size of the file i added). 
+
+2. What is the reason for this problem?
+A copy of the file was stored on the history on the folder /.git/objects with the 'git add file.txt; git commit -m "Add big file"' command
+
+3. How do you remove something from Git history after it is pushed to the remote repository? Which commands would you use? 
+git filter-branch --index-filter "git rm -rf --cached --ignore-unmatch path_to_file" HEAD
```
Comment (denstorti) [inline classes/01class/exercises/c01-git03/zarajoy/c01-git03.md:8]: What's this command doing?

File: classes/01class/exercises/c01-git03/zarajoy/c01-git03.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+Source :https://myopswork.com/how-remove-files-completely-from-git-repository-history-47ed3e0c4c35
+
+4. What are the consequences of the previous action?
+you cannot reverse the removal. you should probably make a back up of the repo somewhere.
```
Comment (denstorti) [inline classes/01class/exercises/c01-git03/zarajoy/c01-git03.md:12]: Other users still have older histories, so they could branch out from there and "save" the work.

But what would they notice and trying to push their changes?

File: classes/01class/exercises/c01-git03/zarajoy/c01-git03.md (hunk 1 above)
Comment (zarajoy) [reply classes/01class/exercises/c01-git03/zarajoy/c01-git03.md:8]: thank you, i will admit i saw the solution given online and did not break it down to understand it. Thank you for prompting me to work it out. 
Translated into plain English the command is saying (from what i understand):
In git (git), Rewrite revision history (filter-branch) and rewrite the index (--index-filter). Remove file from working directory (RM), remove file from staging area (--cached), exit gracefully if the file does not exist (--ignore-unmatch). The file (-f not -rf) we are talking about is is XXX (path_to_file). Do this for the last commit (HEAD).

File: classes/01class/exercises/c01-git03/zarajoy/c01-git03.md (hunk 2 above)
Comment (zarajoy) [reply classes/01class/exercises/c01-git03/zarajoy/c01-git03.md:12]: from what i have read filter-branch is a poor choice especially if other users with their own clones are involved. for a singular local instance it is probably ok?
filter-branch can cause multiple copies of each commit, some with removed information, others without it. 
filter-repo might be better for public repos.

Comment (caiocezart) [reply classes/01class/exercises/c01-git03/zarajoy/c01-git03.md:12]: Can you add that extra information to your answer? :)

File: classes/01class/exercises/c01-git03/zarajoy/c01-git03.md (hunk 1 above)
Comment (caiocezart) [reply classes/01class/exercises/c01-git03/zarajoy/c01-git03.md:8]: Can you add that extra information to your answer?

**Pull request merged**